src/zenml/materializers/path_materializer.py

```diff
@@ -0,0 +1,643 @@
+            if data.is_file() or not os.path.exists(str(data)):
+                # Either it's an original file or we need to load from artifact store
+                if os.path.exists(str(data)) and data.is_file():
+                    file_path = data
+                else:
+                    file_path = self.load(Path)
+
+                # Create a direct file visualization
+                html_content = self._create_single_file_html(file_path)
+                html_uri = os.path.join(self.uri, "file_direct.html")
+                with self.artifact_store.open(html_uri, "w") as f:
+                    f.write(html_content)
+
+                visualizations[html_uri] = VisualizationType.HTML
+                return visualizations
+
+            # Handle directory visualization (existing code)
+            # Load the directory (either from original data or from artifact store)
+            if os.path.exists(str(data)):
+                directory_path = data
+            else:
+                directory_path = self.load(Path)
+
+            # Prepare file data
+            files_data = []
+            for file_path in sorted(directory_path.glob("**/*")):
+                if file_path.is_file():
+                    rel_path = str(file_path.relative_to(directory_path))
+                    size_bytes = file_path.stat().st_size
+
+                    # Read the file content
+                    with open(file_path, "rb") as f:
+                        content_bytes = f.read()
+                    content_b64 = base64.b64encode(content_bytes).decode(
+                        "utf-8"
+                    )
+
+                    # For text files, also get preview content
+                    is_text = self._is_text_file(file_path)
+                    preview_content = ""
+                    if is_text:
+                        preview_content = content_bytes.decode(
+                            "utf-8", errors="replace"
+                        )
+
+                    files_data.append(
+                        {
+                            "path": rel_path,
+                            "filename": os.path.basename(rel_path),
+                            "size": self._format_size(size_bytes),
+                            "size_bytes": size_bytes,
+                            "data": content_b64,
+                            "is_text": is_text,
+                            "preview": preview_content if is_text else "",
+                        }
+                    )
+
+            # Create an HTML page with direct download links
+            html_content = self._create_direct_download_html(files_data)
+            html_uri = os.path.join(self.uri, "files_direct.html")
+            with self.artifact_store.open(html_uri, "w") as f:
+                f.write(html_content)
+
+            visualizations[html_uri] = VisualizationType.HTML
+
+        except Exception as e:
+            # If something goes wrong, create a simple error visualization
```

Comment:
This should either not exist, or exist as a generic materializer functionality or be handled in the frontend.